sensor_test.go

```diff
@@ -0,0 +1,915 @@
+		for _, c := range conn {
+			s, err := SensorFor(c.sensor.address, c.sensor.driver)
+			if err != nil {
+				t.Fatalf("unexpected error: %v", err)
+			}
+			commands, err := s.Commands()
+			if len(c.sensor.commands) == 0 {
+				if err == nil {
+					t.Error("expected error getting commands from non-commandable sensor")
+				}
+				continue
+			}
+			if err != nil {
+				t.Fatalf("unexpected error getting commands: %v", err)
+			}
+			if !reflect.DeepEqual(commands, c.sensor.commands) {
+				t.Errorf("unexpected commands value: got:%q want:%q", commands, c.sensor.commands)
+			}
+			for _, command := range commands {
+				err := s.Command(command).Err()
+				if err != nil {
+					t.Errorf("unexpected error for command %q: %v", command, err)
+				}
+
+				got := c.sensor.lastCommand
+				want := command
+				if got != want {
+					t.Errorf("unexpected command value: got:%q want:%q", got, want)
+				}
+			}
+			for _, command := range []string{"invalid", "another"} {
+				err := s.Command(command).Err()
+				if err == nil {
+					t.Errorf("expected error for trigger %q", command)
+				}
+
+				got := c.sensor.lastCommand
+				dontwant := command
+				if got == dontwant {
+					t.Errorf("unexpected invalid trigger value: got:%q", got, dontwant)
+				}
+			}
+		}
+	})
+
+	t.Run("Binary data", func(t *testing.T) {
+		for _, c := range conn {
+			s, err := SensorFor(c.sensor.address, c.sensor.driver)
+			if err != nil {
+				t.Fatalf("unexpected error: %v", err)
+			}
+			format, err := s.BinDataFormat()
+			if err != nil {
+				t.Fatalf("unexpected error getting modes: %v", err)
+			}
+			if format != c.sensor.binDataFormat {
+				t.Errorf("unexpected bin data format value: got:%q want:%q", format, c.sensor.binDataFormat)
+			}
+			data, err := s.BinData()
+			if err != nil {
+				t.Fatalf("unexpected error getting modes: %v", err)
+			}
+			if !reflect.DeepEqual(data, c.sensor.binData) {
+				t.Errorf("unexpected bin data value: got:% 02x want:% 02x", data, c.sensor.binData)
+			}
+		}
+	})
+
+	t.Run("Direct", func(t *testing.T) {
+		s, err := SensorFor(conn[0].sensor.address, conn[0].sensor.driver)
+		if err != nil {
+			t.Fatalf("unexpected error: %v", err)
+		}
+		f, err := s.Direct(os.O_RDWR)
+		if err != nil {
+			t.Fatalf("unexpected error getting direct connection: %v", err)
+		}
+		defer f.Close()
+		got, err := ioutil.ReadAll(f)
+		if err != nil {
+			t.Fatalf("unexpected error reading direct connection: %v", err)
+		}
+		if !reflect.DeepEqual(got, conn[0].sensor.direct) {
+			t.Errorf("unexpected direct value: got:%q want:%q", got, conn[0].sensor.direct)
+		}
+		err = f.Truncate(0)
+		if err != nil {
+			t.Fatalf("unexpected error seeking to start: %v", err)
+		}
+		_, err = f.Seek(0, io.SeekStart)
+		if err != nil {
+			t.Fatalf("unexpected error seeking to start: %v", err)
+		}
+		new := []byte("new data")
+		_, err = f.Write(new)
+		if err != nil {
+			t.Fatalf("unexpected error seeking to start: %v", err)
```

Comment:
Mismatched error message? Maybe also at other places, I'd have to give it a second pass.